src/main/java/duke/MarkCommand.java

```diff
@@ -0,0 +1,27 @@
+package duke;
+
+import java.io.IOException;
+import java.util.List;
+
+public class MarkCommand extends Command {
+    private int taskNumber;
+
+    /**
+     * Constructor for a Mark command.
+     *
+     * @param taskNumber Task number of the task in the task list to be marked as done.
+     */
+    public MarkCommand(int taskNumber) {
+        this.taskNumber = taskNumber;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        List<Task> taskList = tasks.getTasks();
+        Task currTask = taskList.get(taskNumber - 1);
+
+        currTask.setDone(true);
```

Comment:
There's no need to create a new List instance, I personally feel that these 3 lines of code can be rewritten as:
```
List<Task> taskList = tasks.getTasks();
        Task currTask = taskList.get(taskNumber - 1);

        currTask.setDone(true);
```
```
tasks.get(taskNumber).setDone(true)
```
By implementing a  `get` method in `TaskList`, you can get a reference of the `Task` and call task methods to mark/unmark them.